File: eAnimalShelter/eAnimalShelter.Model/Requests/UserInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace eAnimalShelter.Model.Requests
{
    public class UserInsertRequest
{
    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Username { get; set; } = string.Empty;

    public string Password { get; set; } = string.Empty;

    public string? PhoneNumber { get; set; }

    public string Address { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public int RoleId { get; set; } 
}
}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/requests/user_insert_request.dart
class UserInsertRequest {
  String firstName;
  String lastName;
  String email;
  String username;
  String password;
  String? phoneNumber;
  String address;
  bool isActive;
  int roleId;

  UserInsertRequest({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.username,
    required this.password,
    this.phoneNumber,
    required this.address,
    required this.roleId,
    this.isActive = true,
  });

  Map<String, dynamic> toJson() {
    return {
      "firstName": firstName,
      "lastName": lastName,
      "email": email,
      "username": username,
      "password": password,
      "phoneNumber": phoneNumber,
      "address": address,
      "isActive": isActive,
      "roleId": roleId,
    };
  }
}